src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -138,7 +138,15 @@ protected function compileAggregate(Builder $query, $aggregate)
             $column = 'distinct '.$column;
         }
 
-        return 'select '.$aggregate['function'].'('.$column.') as aggregate';
+        $sql = 'select ';
+
+        if ($query->groups) {
+            $sql .= $this->columnize($query->groups).', ';
```

Comment:
This adds the group columns in the results, even if they are not used.
If a group column is named `aggregate`, the DB engine will prefix it with the table name. We don't have conflict with the calculated `aggregate`.

Reply:
I have fear this is going to be another breaking change 😅 any way we can be sure?

Reply:
This doesn't affect queries created by the query builder. The result of the aggregate functions `count`, `max`... will stay the same (https://github.com/laravel/framework/issues/53581).

It only add fields to the result when calling the grammar method directly. I moved the new columns last to keep the `aggregate` result first.